rotkehlchen/globaldb/cache.py

```diff
@@ -1,71 +1,84 @@
 """Functions dealing with the general_cache table of the Global DB"""
 from collections.abc import Iterable
 from typing import Optional, Union
+
 from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
 from rotkehlchen.chain.evm.types import string_to_evm_address
-
 from rotkehlchen.db.drivers.gevent import DBCursor
 from rotkehlchen.types import ChecksumEvmAddress, GeneralCacheType, Timestamp
 from rotkehlchen.utils.misc import ts_now
 
+UNIQUE_CACHE_TABLE_NAME = 'unique_cache'
+GENERAL_CACHE_TABLE_NAME = 'general_cache'
+UNIQUE_CACHE_KEYS: set[GeneralCacheType] = {
+    GeneralCacheType.CURVE_POOL_ADDRESS,
+    GeneralCacheType.MAKERDAO_VAULT_ILK,
+    GeneralCacheType.CURVE_GAUGE_ADDRESS,
+}
+
 
-def compute_cache_key(key_parts: Iterable[Union[str, GeneralCacheType]]) -> str:
+def compute_cache_key(key_parts: Iterable[Union[str, GeneralCacheType]]) -> tuple[str, str]:
     """Function to compute cache key before accessing globaldb cache.
     Computes cache key by iterating through `key_parts` and making one string from them.
-    Only tuple with the same values and the same order represents the same key."""
+    Only tuple with the same values and the same order represents the same key. Also returns which
+    cache table the data must me stored in."""
+    unique_cache = list(key_parts)[0] in UNIQUE_CACHE_KEYS
+    cache_table = UNIQUE_CACHE_TABLE_NAME if unique_cache else GENERAL_CACHE_TABLE_NAME
     cache_key = ''
     for part in key_parts:
         if isinstance(part, GeneralCacheType):
             cache_key += part.serialize()
         else:
             cache_key += part
-    return cache_key
+    return cache_key, cache_table
 
 
 # Using any random address here, since length of all addresses is the same
-BASE_POOL_TOKENS_KEY_LENGTH = len(compute_cache_key([GeneralCacheType.CURVE_POOL_TOKENS, ZERO_ADDRESS]))  # noqa: E501
+BASE_POOL_TOKENS_KEY_LENGTH = len(compute_cache_key([GeneralCacheType.CURVE_POOL_TOKENS, ZERO_ADDRESS])[0])  # noqa: E501
 
 
-def globaldb_set_general_cache_values_at_ts(
+def globaldb_set_cache_values_at_ts(
         write_cursor: DBCursor,
         key_parts: Iterable[Union[str, GeneralCacheType]],
-        values: Iterable[str],
+        values: Union[str, Iterable[str]],
         timestamp: Timestamp,
 ) -> None:
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it."""
-    cache_key = compute_cache_key(key_parts)
+    cache_key, cache_table = compute_cache_key(key_parts)
+    if isinstance(values, str):
```

Comment:
I get it why you do the if check here but it's an unecessary if check which can probably be coded around by either adding morefunctions or just making the caller pass it as an iterable with a single value.

Reply:
it is how the function was originally, but was advised to add an overload or  something of such in case of single string